cedar-policy-core/src/est/err.rs

```diff
@@ -34,6 +34,9 @@ pub enum FromJsonError {
     #[error("tried to convert JSON representing a template to a static policy")]
     #[diagnostic(transparent)]
     TemplateToPolicy(#[from] ast::UnexpectedSlotError),
+    /// Tried to convert an EST representing a static policy to an AST representing a template
+    #[error("tried to convert JSON representing a static policy to a template")]
+    PolicyToTemplate,
```

Comment:
just for consistency, even though this isn't a public error type (I think?), should we make `struct PolicyToTemplateError {}` and have this be `PolicyToTemplate(#[from] PolicyToTemplateError)`?

Reply:
Done in the latest commit. Also updated the CST->AST path to use the same error structure for consistent messaging